### ISSUE TITLE
Date of last contact (customer) - make behaviour configurable

### ISSUE THREAD

**Issue by waggonservice** [zammad/zammad#1865]
### Infos:

* Used Zammad version: 2.3.x
* Installation method (source, package, ..):
* Operating system:
* Database + version:
* Elasticsearch version:
* Browser + version:

### Expected behavior:

* Every incoming mail from costumers sets the date of the data-field "last contact (customer)".

### Actual behavior:

* The customer (especially its company) who started the ticket sets the date "last contact (customer)". If someone from another company sends a mail to this ticket, the date won't be changed. 
In Overviews: The data-field "Updated at" shows the update of the ticket, but it is interpreted by agents as a work in progress and not as a new incoming mail. New information get lost.

### Steps to reproduce the behavior:

*  
1) customer1@company1.com started a new ticket
2) customer2@company1.com sent a mail to the same ticket -> set the new date
3) customer3@company2.com sent a mail to the same ticket -> won't set the new date in the data-field "last contact (customer)"

Please take a look at the attached figures.

In conclusion: It is important to make a difference between an update of a ticket by a customer or by an agent. The data-field "last contact (customer)" doesn't work as expected.

(Yes I'm sure this is a bug and no feature request or a general question.)

![overview](https://user-images.githubusercontent.com/31927348/37092066-a95fc714-220a-11e8-8da4-c3703651c55e.png)
![ticket](https://user-images.githubusercontent.com/31927348/37092067-a9801c3a-220a-11e8-93ff-ae87e2d2363d.png)

**Comment by martini** [comment]
Hi @waggonservice 

I'm not 100% sure, what the expected behaviour is, so I need to clarify the current implementation:

**1) No differentiate between customers**
Currently Zammad does not differ if the sender is set as a customer on the ticket or not. At present, only the role of the sender is decisive. 

If the created article has been categorized as sender from customer - no matter which customer (you can see this at the position of the avatar on the article), then "last_contact_customer" is updated.

**2) What if customer sends 2 times**
If a customer (no matter which customer) it sending two emails to Zammad where no agent response was between, the last_contact_customer is only updated at first email.

The reason for this is the following use case:
- Overview: Show me all tickets where I need to reply first (based on last contact of the customer)

So a little example: 
1) customer is creating a ticket
2) 2 days later the customer sends an complaint because nobody has responding

 -> Then the last_contact_customer will still have the timestamp of the 1 customer article. Because otherwise the ticket in the overview is not sorted on top. And because the customer has complained, the response is sent even later.

==> Finally the question, did you expect that the last_contact_customer should also get updated if two customer articles are created without agent response between? 

If you, just let me know your use case to understand it. Thanks a lot!

 -Martin

**Comment by waggonservice** [comment]
Hello Martin,

ok now I understand the behaviour. It does make sense that zammad shows dates of non-answered mails.

I want to give you an example where this behaviour is maybe problematic:

* The customer sends a mail like "I will give you additional information in a few hours."
* The agent registered this mail but there is no need to answer.
* The customer sends 5 hours later additional information, but the date won't change. (see attachement)

==> If the agent has many tickets, it is possible to miss this ticket. (Reminders are helpful, but many reminders are problematic too)

If there is a data-field that shows the date of the last incoming mail, you are able to sort in an overview by this data-field and get a well structured inbox.

![overview](https://user-images.githubusercontent.com/31927348/37101515-123b5706-2226-11e8-8af2-bb87cfc72219.png)
![ticket 2](https://user-images.githubusercontent.com/31927348/37101514-1218ace2-2226-11e8-873a-73765d9fd471.png)

**Comment by gwingend** [comment]
Hi, @martini and @waggonservice ,

I think I would agree to both of you.
The argumentation of @martini concerning customer complains is right.
For our site this would be the usecase we probably have most. So I like the actual behaviour.

But the argumentation of @waggonservice concerning additional information is a usecase too.

So for me, depending on the company, both behaviours are valid usecases.
But I think there is only one option possible.

Maybe the best solution would be a configuration switch.
So the admin of the zammad installation could configure the behaviour in the way he likes most.

**Comment by waggonservice** [comment]
Thank you @gwingend for your support.

In our usecase we can't sort by the oldest date of last contact (customer). We need often additional information from customers and third parties and work parallel on many tickets. In that case it would be useful to sort by last update from customer to notice updated tickets.

In our opinion it would be a useful data-field. What do you think @martini ?

**Comment by martini** [comment]
@waggonservice as both of you are paying customers we will add a config option for that. I'll reply as soon as possible.

**Comment by muhammadn** [comment]
I have actually written the code which is still in QA to add this feature.

Under 'Settings' -> 'Ticket', you have and option to select 'Ticket Last Contact Behaviour' to enable or disable the update of the customer last contact.

**Comment by martini** [comment]
In upcoming Zammad version you can change this behaviour in Admin -> Setting -> Ticket -> 'Ticket Last Contact Behaviour'. All new created or updated tickets will have he new behaviour for the  last_contact_customer attribute.

Feedback is welcome.
